### PR TITLE
Clean up and booleans between segments, volumes and surfaces. Improved c++ exception localization.

### PR DESCRIPTION
**Pull request by pentacular**

### CODE REVIEW

File: algorithm/cgal/arrangeSegments.js [hunk 1]
```diff
@@ -10,46 +10,59 @@ export const arrangeSegments = (
...
-      c.addPoint(out, start[X], start[Y], start[Z]);
-      c.addPoint(out, end[X], end[Y], end[Z]);
-    }
-  };
-  const emitPolygon = (isHole) => {
-    target = { points: [], exactPoints: [], holes: [], plane, exactPlane };
-    if (isHole) {
-      polygon.holes.push(target);
+  try {
+    const c = getCgal();
```
Comment (BarbourSmith) [inline algorithm/cgal/arrangeSegments.js:14]: getCgal() returns a new instance, right? It's not a singleton?

File: algorithm/cgal/build_node.sh [hunk 2]
```diff
@@ -2,4 +2,8 @@
 
 clang-format --style=google -i cgal.cc
 
-emcc -DCGAL_ALWAYS_ROUND_TO_NEAREST -DCGAL_WITH_GMPXX -DCGAL_USE_GMPXX=1 -DCGAL_DO_NOT_USE_BOOST_MP -DBOOST_ALL_NO_LIB cgal.cc -I . -I ~/opt/include -L ~/opt/lib -static -O3 -std=c++1z --bind -o cgal_node.js -lgmpxx -lmpfr -lgmp -s MODULARIZE=1 -s USE_ES6_IMPORT_META=0 -s USE_BOOST_HEADERS=1 -s ALLOW_MEMORY_GROWTH=1 -s ASSERTIONS=1 -s DISABLE_EXCEPTION_CATCHING=0 -ferror-limit=10000 && mv cgal_node.js cgal_node.cjs
+emcc -Wunused -Wunused-function -DCGAL_ALWAYS_ROUND_TO_NEAREST -DCGAL_WITH_GMPXX -DCGAL_USE_GMPXX=1 -DCGAL_DO_NOT_USE_BOOST_MP -DBOOST_ALL_NO_LIB cgal.cc -I . -I ~/opt/include -L ~/opt/lib -static -O3 -std=c++1z --bind -o cgal_node.js -lgmpxx -lmpfr -lgmp -s MODULARIZE=1 -s USE_ES6_IMPORT_META=0 -s USE_BOOST_HEADERS=1 -s ALLOW_MEMORY_GROWTH=1 -s ASSERTIONS=1 -s DISABLE_EXCEPTION_CATCHING=0 -ferror-limit=10000 && mv cgal_node.js cgal_node.cjs
```
Comment (BarbourSmith) [inline algorithm/cgal/build_node.sh:5]: Man this is really getting long and complex 😛

File: nb/projects/cnc/spindle_mount.nb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Arc(44.5)
```
Comment (BarbourSmith) [inline nb/projects/cnc/spindle_mount.nb:1]: 😁

**Review by BarbourSmith — APPROVED** [review]
This was a huge one! Very nice. No big thoughts on my end, looks good to me 😄

File: algorithm/cgal/arrangeSegments.js (hunk 1 above)
Comment (pentacular) [reply algorithm/cgal/arrangeSegments.js:14]: It returns a singleton, allocated on the first call.

**Pull request merged**